cmd/kubeadm/app/cmd/reset.go

```diff
@@ -131,6 +133,20 @@ func (r *Reset) Run(out io.Writer, client clientset.Interface) error {
 	// Only clear etcd data when using local etcd.
 	etcdManifestPath := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ManifestsSubDirName, "etcd.yaml")
 
+	cfg, err := configutil.FetchConfigFromFileOrCluster(client, os.Stdout, "reset", "", false)
+
+	if err != nil && cfg != nil {
+		cfg = nil
+	}
+
+	// Check if control plane node
+	klog.V(1).Infof("[reset] checking if control plane node")
+	if isControlPlane(cfg) {
+		klog.V(1).Infof("[reset] removing control plane node from cluster")
+		uploadconfig.UploadRemoveConfiguration(cfg, client)
+	}
+
+	// Try to stop etcd
 	klog.V(1).Infof("[reset] checking for etcd config")
 	etcdDataDir, err := getEtcdDataDir(etcdManifestPath, client)
```

Comment:
The only thing this needs the `client` for is to get the config..... what I did above? seems like a waste for it to do it again... but I did not have to change this code so I left it.